udp.go

```diff
@@ -9,6 +9,34 @@ import (
 	"golang.org/x/net/ipv6"
 )
 
+type SessionUDPFactory interface {
+	// InitConn set's up 'conn' to be used with a SessionUDP.
+	// Must be called before 'conn' is passed to SessionUDP.ReadRequest()
+	InitConn(conn *net.UDPConn) error
```

Comment:
This method isn't needed either, just use `ActivateAndServe` instead of `ListenAndServe` and pass in your own `net.UDPConn` with whatever options you need set.

Reply:
`ActivateAndServe` also sets UDP socket options, would there be a way to bypass that?

Reply:
Oh it does too. Are the socket options that are set by default a problem for you? If so, it would be easier to just hang a setting off of `Server` to disable that. (I’ll look into that once I’ve got a better understanding of the existing limitations).